include/remill/Arch/Runtime/Types.h

```diff
@@ -64,12 +65,56 @@ static_assert(4 == sizeof(float32_t), "Invalid `float32_t` size.");
 typedef double float64_t;
 static_assert(8 == sizeof(float64_t), "Invalid `float64_t` size.");
 
+#if defined(__x86_64__) || defined(__i386__) || defined(_M_X86)
+typedef long double float128_t;
+static_assert(12 == sizeof(float128_t) || 16 == sizeof(float128_t), "Invalid `float128_t` size.");
+#else
 typedef double float128_t;
 static_assert(8 == sizeof(float128_t), "Invalid `float128_t` size.");
+#endif
+
 
-// TODO(pag): Assumes little endian.
 struct float80_t final {
-  uint8_t data[10];
+
+  // TODO: float80_t is backed by float64_t since we would
+  //       not like to deal with converting long double to
+  //       extended-precision format. It will be error prone
+  //       and may pollute semantics.
+  //
+  union{
+    uint16_t data[5];
+    struct {
+      float64_t f64;
```

Comment:
I am using `fp64` as the backing for `float80_t`. This is because cutting the padding bits from long double causes most of the test cases to fail. In some cases the ST registers output mismatch and in other cases the FPU exception flags are not getting set correctly. 

The FPU status flag differences are due to the extended-precision format. And some of them can be resolved by fixing the utility functions.

Reply:
One approach is to extend `float80_t` to the size of `long double` and let extra bits run into the paddings. This will not require cutting off the bits and causing issues. @pgoodman might be suggesting the same approach earlier.